### ISSUE TITLE
Problems with nloptr 1.0.4 and R 3.3.0

### ISSUE THREAD

**Issue by fgomesbarros** [astamm/nloptr#21]
Hi,

I can't install the package nloptr 1.0.4 on R 3.3.0. It always appears the following message:

```
> install.packages("nloptr")
Installing package into ‘/Users/fgomesbarros/Library/R/3.3/library
(as ‘lib’ is unspecified)
trying URL 'https://cran.revolutionanalytics.com/src/contrib/nloptr_1.0.4.tar.gz'
Content type 'application/octet-stream' length 353942 bytes (345 KB)
==================================================
downloaded 345 KB

* installing *source* package ‘nloptr’ ...
** package ‘nloptr’ successfully unpacked and MD5 sums checked
checking for g++... g++
checking whether the C++ compiler works... yes
checking for C++ compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking how to run the C++ preprocessor... g++ -E
checking whether we are using the GNU C++ compiler... (cached) yes
checking whether g++ accepts -g... (cached) yes
checking for pkg-config... yes
configure: Now testing for NLopt header file.
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking nlopt.h usability... yes
checking nlopt.h presence... yes
checking for nlopt.h... yes
configure: Suitable NLopt library found.
configure: creating ./config.status
config.status: creating src/Makevars
** libs
/usr/bin/clang++ -I/opt/local/Library/Frameworks/R.framework/Resources/include -DNDEBUG  -I/opt/local/include    -fPIC  -pipe -Os -arch x86_64 -stdlib=libstdc++  -c dummy.cpp -o dummy.o
/usr/bin/clang -I/opt/local/Library/Frameworks/R.framework/Resources/include -DNDEBUG  -I/opt/local/include    -fPIC  -pipe -Os -arch x86_64  -c nloptr.c -o nloptr.o
/usr/bin/clang++ -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress -L/opt/local/Library/Frameworks/R.framework/Resources/lib -L/opt/local/lib -Wl,-headerpad_max_install_names -o nloptr.so dummy.o nloptr.o -F/opt/local/Library/Frameworks/R.framework/.. -framework R -Wl,-framework -Wl,CoreFoundation
installing to /Users/fgomesbarros/Library/R/3.3/library/nloptr/libs
** R
** inst
** preparing package for lazy loading
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded
Error in dyn.load(file, DLLpath = DLLpath, ...) : 
  unable to load shared object '/Users/fgomesbarros/Library/R/3.3/library/nloptr/libs/nloptr.so':
  dlopen(/Users/fgomesbarros/Library/R/3.3/library/nloptr/libs/nloptr.so, 6): Symbol not found: _nlopt_add_equality_mconstraint
  Referenced from: /Users/fgomesbarros/Library/R/3.3/library/nloptr/libs/nloptr.so
  Expected in: flat namespace
 in /Users/fgomesbarros/Library/R/3.3/library/nloptr/libs/nloptr.so
Error: loading failed
Execution halted
ERROR: loading failed
* removing ‘/Users/fgomesbarros/Library/R/3.3/library/nloptr’
* restoring previous ‘/Users/fgomesbarros/Library/R/3.3/library/nloptr’
Warning in install.packages :
  installation of package ‘nloptr’ had non-zero exit status

The downloaded source packages are in
    ‘/private/var/folders/rz/dp6vmb39165406h9qv69q60c0000gn/T/RtmpLvLOmn/downloaded_packages’`
```

My R version is 3.3.0 as you can see below:

```
> version
               _                           
platform       x86_64-apple-darwin12.6.0   
arch           x86_64                      
os             darwin12.6.0                
system         x86_64, darwin12.6.0        
status                                     
major          3                           
minor          3.0                         
year           2016                        
month          05                          
day            03                          
svn rev        70573                       
language       R                           
version.string R version 3.3.0 (2016-05-03)
nickname       Supposedly Educational      
```

Is there any tip to solve it?

I also tried to install in olders versions (2.3.3 and 2.3.4) of R and had the same issue.

Thanks in advance,
Fabio

```
```

**Comment by BruceZhaoR** [comment]
@fgomesbarros  Did you solve it? I encounter the same problem on Ubuntu 16.04.
 ````
Error in dyn.load(file, DLLpath = DLLpath, ...) : 
  unable to load shared object 'Library/R/library/nloptr/libs/nloptr.so':
...

> session_info()
Session info ---------------------------------------------------------------------------------
 setting  value                       
 version  R version 3.3.2 (2016-10-31)
 system   x86_64, linux-gnu           
 ui       RStudio (1.0.136)           
 language (EN)                        
 collate  zh_CN.UTF-8                 
 tz       Asia/Shanghai               
 date     2017-03-06    
````

@jyypma  please CC .

**Comment by BruceZhaoR** [comment]
Solved~ 😃   

a line of command  `sudo apt-get install libnlopt-dev` will solve the problem.
details: <http://stackoverflow.com/questions/29716857/installing-nloptr-on-linux>

**Comment by burcakotlu** [comment]
Unfortunately `sudo apt-get install libnlopt-dev`gave me this error 
```
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: The package apport needs to be reinstalled, but I can't find an archive for it.

```
My question can be reached out at
https://askubuntu.com/questions/1006262/i-can-not-install-anything-using-sudo-apt-get-install/1006269?noredirect=1#comment1628703_1006269

**Comment by eddelbuettel** [comment]
Unrelated:  `E: The package apport needs to be reinstalled,`  means you have a different problem on your system.  Get that sorted out (local help, maybe?) then install `libnlopt-dev`.

**Comment by astamm** [comment]
We just released `v2.0.0`. We now rely on `Cmake` when building from source. Can you test and tell us if it fixes whatever issue you had in the past please?
